fixtures/tests/_fixtures/test_monkeypatch.py

```diff
@@ -20,9 +20,20 @@
 reference = 23
 
 class C(object):
+    def foo(self): pass
     @staticmethod
-    def foo(): pass
-def bar(): pass
+    def foo_staticmethod(): pass
+    @classmethod
+    def foo_classmethod(cls): pass
+
+def fake(arg): pass
+def fake2(arg): pass
+def fake_no_args(): pass
+@staticmethod
```

Comment:
This doesn't make sense to me - staticmethod and classmethod only make sense to adapt how methods receive their parameters - they implement descriptors : so fake_staticmethod will be a descriptor, which can only be used on a class itself. I haven't gotten through all the code yet, but if this is then being bound onto a class, that could work - but perhaps we should take a step back and decide what the UI for all this should be, and work forward from there - because a staticmethod defined on a class and then used to patch a regular method is - at best - odd :)